### PR TITLE
feat(NODE-6157): add signature to github releases

### PR DESCRIPTION
**Pull request by aditi-khare-mongoDB**
### Description
Sign releases in both 5.x and 6.x

#### What is changing?
Automate release signing with a detached signature and verification instructions in the README.
[Link to an example of release signatures working with v2](https://github.com/mongodb/js-bson/actions/runs/9352138423/job/25739562707) (if you check the artifact link at the end of the compress_sign step, you can see the signature)

##### Is there new documentation needed for these changes?
No

#### What is the motivation for this change?
SSDLC Compliance

### Add Signature to Github Releases
The Github release for the `mongodb` package now contains a detached signature file for the NPM package (named
`mongodb-X.Y.Z.tgz.sig`), on every major and patch release to 6.x and 5.x.  To verify the signature, follow the instructions in the 'Release Integrity' section of the `README.md` file.

### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .github/actions/compress_sign_and_upload/action.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      description: 'Artifactory password input for drivers-github-tools/garasign/gpg-sign'
+      required: true
+
+runs:
+  using: composite
+  steps:
+    - run: npm pack
+      shell: bash
+
+    - name: Get release version and release package file name
```
Comment (alcaeus) [inline .github/actions/compress_sign_and_upload/action.yml:24]: This looks generic enough where it's useful for other packages as well. We can consider extracting this into a separate action in the [Github Tools Repository](https://github.com/mongodb-labs/drivers-github-tools) for easier sharing across node repositories.

File: .github/actions/compress_sign_and_upload/action.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      id: vars
+      shell: bash
+      run: |
+        package_version=$(jq --raw-output '.version' package.json)
+        echo "package_version=${package_version}" >> "$GITHUB_OUTPUT"
+        echo "package_file=mongodb-${package_version}.tgz" >> "$GITHUB_OUTPUT"
+    - name: Create detached signature
+      uses: mongodb-labs/drivers-github-tools/garasign/gpg-sign@v1
```
Comment (alcaeus) [inline .github/actions/compress_sign_and_upload/action.yml:32]: Just a heads up that we'll be releasing v2 of the action soon, which will change the way things are set up. There are no plans to deprecate support for v1 anytime soon, so there's no requirement to upgrade right away.

File: .github/actions/compress_sign_and_upload/action.yml (hunk 1 above)
Comment (aditi-khare-mongoDB) [reply .github/actions/compress_sign_and_upload/action.yml:24]: yep, we'd just have to add a package_name field to the inputs of this file, and add it into `package_file`

(didn't realize I didn't send this comment, was in "pending" for 5 days)

**Review by W-A-James — APPROVED** [review]

**Review by W-A-James — DISMISSED** [review]

File: .github/actions/compress_sign_and_upload/action.yml (hunk 2 above)
Comment (baileympearson) [reply .github/actions/compress_sign_and_upload/action.yml:32]: Since we have the credentials and setup now, can we use v2 instead?

**Review by baileympearson — CHANGES_REQUESTED** [review]

**Comment by baileympearson** [comment]
Do we have an example of a successful signing using the gpg-sign v2 action?  Either here or from js-bson would be fine since I assume the scripts are almost identical.

**Comment by aditi-khare-mongoDB** [comment]
@baileympearson Just added a link to an example in the PR description!

**Review by baileympearson — APPROVED** [review]

**Review by W-A-James — APPROVED** [review]

**Pull request merged**